docs/proposals/central-segmented-registry.md

```diff
@@ -0,0 +1,144 @@
+          apiVersion: v1
+          metadata:
+            labels:
+              app: docker-registry
+            name: docker-registry
+          spec:
+            tls:
+              ...
+            to:
+              kind: Service
+              name: docker-registry
+          status: {}
+
+An Endpoint object created in the "default" project namespace on each cluster. In this example, the test OpenShift cluster references the test registry instance in the central registry cluster.
+
+        kind: Endpoints
+        apiVersion: v1
+        metadata:
+          name: docker-registry
+        subsets:
+          - addresses:
+              - IP: docker-registry-test.apps.example.com
+            ports:
+              - port: 443
+                name: docker-registry
+
+### Managing Authorization
+
+Cross-cluster authorization is a significant challenge. Automating or synchronizing cross-cluster authorization is ultimately desired. Until that can be achieved, manual processes may need to bridge the gap.
+
+**Deployments**
+
+Initially, a cluster administrator configures a service account with pull access to the appropriate registry environment so deployments "just work". This service account would have access to *all images* in the central registry irregardless of the environment. We are relying on each cluster registry service object to prevent pulling images from unintended environments. Additional engineering work would be required if restricting pulling images is required.
+
+**Builds**
+
+Build and push authorization is very important. Certain users or projects on certain clusters need access to certain registries. Initially, users could retrieve a "builder" service account token from the registry cluster for their namespace and create a *.dockercfg* build secret. This entails managing namespaces on two clusters and is not ideal.
+
+For organizations that maintain separate build clusters, the registry cluster could be part of this infrastructure cluster. This solves several authorization and namespace challenges.
+
+**Image Scans**
+
+As we look towards implementing integrated image scanning, the central registry cluster is a good candidate for running scanning workloads that check for vulnerabilities at the registry.
+
+## Background
+
+### Actions referencing registries in some way
+
+* deploy -> authenticated pull from registry
+* build -> authenticated push to registry
+* tag -> updating image metadata, importing image
+
+### Image Objects that reference registry in some way [FIXME]
+
+#### Images
+
+* ImageStreamImage
+* ImageStreamMapping
+* ImageStreamTag
+* ImageSource
+
+#### Other Objects
+
+* BuildConfigs
+* DeploymentConfigs
+
+## Questions and Challenges
+
+1. Should registry abstraction be a first-class object? How might this help?
+1. Can we delete a string from an imageStreamTag list?
+1. How can we make manipulating images from the cluster (not the registry cluster) feel native? How might image pull-through help?
```

Comment:
I'm more concerned with users getting confused with where to manage images. What happens if you tag an imagestream on the test cluster but not the registry cluster? That seems like a big problem. At least I haven't completely thought it through.